exporter/opensearchexporter/sso_log_exporter.go

```diff
@@ -60,16 +65,100 @@ func (l *logExporter) Start(ctx context.Context, host component.Host) error {
 }
 
 func (l *logExporter) pushLogData(ctx context.Context, ld plog.Logs) error {
-	indexer := newLogBulkIndexer(l.Index, l.bulkAction, l.model)
+	indexer := newLogBulkIndexer("", l.bulkAction, l.model)
 	startErr := indexer.start(l.client)
 	if startErr != nil {
 		return startErr
 	}
+
+	// Collect attributes from resource/log record
+	attrs := collectAttributes(ld)
+	logTimestamp := time.Now() // Replace with actual log timestamp extraction
+
+	// Index name resolution
+	// If LogsIndex is not set, use the default index name, otherwise resolve it using the config and attributes.
+	// This allows for dynamic index naming based on log attributes.
+	var indexName string
+	if l.config.LogsIndex == "" {
+		indexName = l.Index
+	} else {
+		indexName = resolveLogIndexName(l.config, attrs, logTimestamp)
+	}
+	indexer.index = indexName
 	indexer.submit(ctx, ld)
 	indexer.close(ctx)
 	return indexer.joinedError()
 }
 
+// collectAttributes extracts resource and log record attributes into a flat map for placeholder resolution.
+func collectAttributes(ld plog.Logs) map[string]string {
+	attrs := make(map[string]string)
+	resLogsSlice := ld.ResourceLogs()
+	for i := 0; i < resLogsSlice.Len(); i++ {
+		resLogs := resLogsSlice.At(i)
+		// Resource attributes
+		resAttrs := resLogs.Resource().Attributes()
+		resAttrs.Range(func(k string, v pcommon.Value) bool {
+			attrs[k] = v.AsString()
+			return true
+		})
+		logSlice := resLogs.ScopeLogs()
+		for j := 0; j < logSlice.Len(); j++ {
+			scopeLogs := logSlice.At(j)
+			// Instrumentation scope attributes
```

Comment:
Is it intentional that you ignore the instrumentation scope attributes?

Reply:
No, we need to include scope log attributes as well. Thanks!

Reply:
Thanks @paulstn for adding this update.